### PR TITLE
Leaflet: Replaced PhantomJS Dependency with Firefox

### PR DESCRIPTION
**Pull request by ossdev07**
PhantomJS is an unmaintained project, Replaced phantomjs with firefox in karma.conf.js.
Increased the capture timeout to ```10000``` ms as for aarch64 it is not able to capture the browser in ```5000``` ms.
Updated the travis.yml as well to run the package with firefox.

### CODE REVIEW

**Comment by cherniavskii** [comment]
Hi,
While I agree we should get rid of unmaintained PhantomJS, I'd prefer to replace it with other headless runner, since it doesn't require to have specific browser installed

**Comment by ossdev07** [comment]
Sure @cherniavskii I agree with you. 

I have checked the package with firefox browser in the headless environment if you agree I can update the PR with the required changes.

**Comment by ossdev07** [comment]
@cherniavskii Please review the changes I have added the firefox headless support in the package.

**Comment by cherniavskii** [comment]
Hi @ossdev07 
Thanks for taking your time updating a PR.
I've checked out your branch, but it requires Firefox to be installed.
I'd rather keep PhantomJS for now, and eventually replace it with puppeteer, which downloads chromium binary automatically.

This is blocked by failing tests though (see #5845).
I'm gonna try to fix those test, or skip them in Chrome (they're failing for a long time anyway).

Would you like to give puppeteer a try?

As for CI - we already have tests running in Firefox in Travis CI.

**Comment by ossdev07** [comment]
@cherniavskii I do agree with you that for using firefox headless, firefox binary needs to be downloaded.

But the puppeteer does not have support for all the architectures i.e it does not support aarch64 because the chromium binary which gets downloaded at puppeteer/.local-chromium/linux-637110/chrome-linux is of x86-64 because of which the package does not support aarch64.

```
 file chrome
chrome: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, not stripped
```

For providing multi-architecture support for the package I was suggesting to use firefox headless, please do share your thoughts on this.

**Comment by johnd0e** [comment]
> Updated the travis.yml as well to run the package with firefox.

As I see from upper discussion, Leaflet maintainers are not ready to replace current testing flow yet.

But may be we can facilitate adoption of this PR (and/or #6852), if we keep PhantomJS, but also setup alternative testing frameworks in parallel, as optional explicit npm-commands.

@cherniavskii @ossdev07 ?

**Comment by ossdev07** [comment]
@johnd0e If it seems more feasible to use phantomjs along with the other testing frameworks then we can use phantomjs as an optional dependency too. Please suggest what are your thoughts on this.

**Comment by johnd0e** [comment]
Well, one of available frameworks still need to be main. 
I propose to adopt new ones in steps, e.g.: 
- set new as optional, to allow everyone to test them thoroughly
- eventually, after leaflet maintainers will be ready, they'll make it as main

File: spec/karma.conf.js [hunk 1]
```diff
@@ -89,24 +88,24 @@ module.exports = function (config) {
...
+                                flags: ['--headless'],
+                                options: {
+                                        onCallback: function (data) {
+                                                if (data.render) {
+                                                        page.render(data.render);
+                                                }
+                                        }
+                                }
+                        }
+                },
```
Comment (johnd0e) [inline spec/karma.conf.js:105]: Why do you need this?

File: .travis.yml [hunk 2]
```diff
@@ -2,6 +2,10 @@ sudo: false
 language: node_js
 node_js:
   - node
+before_install:
+  - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x16"
```
Comment (johnd0e) [inline .travis.yml:6]: Is it really needed in headless mode?

**Comment by johnd0e** [comment]
See #7115

**Comment by mourner** [comment]
Superceded by #7115

**Pull request closed without merging**